File: longestsubstring.py
#this is my solution for lenght of longest substring on leetcode

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        ans = set()
        start = 0
        end = 0
        maxlength = 0
        
        while end < len(s):
            if s[end] not in ans:
                ans.add(s[end])
                end += 1
                maxlength = max(maxlength, end - start)
            else:
                ans.remove(s[start])
                start += 1
        
        return maxlength

File: longestsubstring.cpp
//this is my leetcode solution for longest substring in a string

class Solution {
public:
  int lengthOfLongestSubstring(string s) {
    std::unordered_set<char> ans;
    int start = 0, end = 0, maxlength = 0;
    while (end < s.size()) {
      if (ans.find(s[end]) == ans.end()) {
        ans.insert(s[end]);
        ++end;
        maxlength = std::max(maxlength, end - start);
      } else {
        ans.erase(s[start]);
        ++start;
      }
    }
    return maxlength;
  }
};
